### PR TITLE
Add model, action, selector, reducer

### PR DESCRIPTION
**Pull request by ssharif6**
Part of #634

Note: UI code not covered in this PR, just state changes, action, reducer.

* Add logic to toggle hiding related-views
* Added flag in State to represent if related-views are hidden/unhidden
* Add tests for related-view reducer
* Add missing default state and add missing reducer to combineReducers

### CODE REVIEW

File: src/actions/related-views.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import {PlainReduxAction} from './redux-action';
+
+export type RelatedViewAction = HideRelatedView;
+
+export const RELATED_VIEW_HIDE_TOGGLE = 'RELATED_VIEW_HIDE_TOGGLE';
```
Comment (kanitw) [inline src/actions/related-views.ts]: This is hiding _all_ related view**s** so I wonder if the action should be `RELATED_VIEWS_HIDE_TOGGLE` instead. 

(Similar for `RelatedViewAction` -- however, we may have action that is specific to one kind of related views in the future so I have less strong opinion about the `RelatedViewAction`.)

File: src/models/index.ts [hunk 2]
```diff
@@ -22,6 +23,7 @@ export interface PersistentState {
   bookmark: Bookmark;
   config: VoyagerConfig;
   log: Log;
+  relatedViewToggler: RelatedViewToggler;
```
Comment (kanitw) [inline src/models/index.ts]: Perhaps just `relatedViews` -- as we may have other related views feature in the future that are not boolean / togglers.

File: src/models/relatedViewToggler.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+
+export interface RelatedViewToggler {
+  isRelatedViewHidden: boolean;
```
Comment (kanitw) [inline src/models/relatedViewToggler.ts]: hidden: boolean

File: src/models/index.ts [hunk 4]
```diff
@@ -64,6 +66,7 @@ export const DEFAULT_PERSISTENT_STATE: PersistentState = {
   bookmark: DEFAULT_BOOKMARK,
   config: DEFAULT_VOYAGER_CONFIG,
   log: DEFAULT_LOG,
+  relatedViews: DEFAULT_RELATED_VIEW_TOGGLER,
```
Comment (kanitw) [inline src/models/index.ts]: DEFAULT_RELATED_VIEWS

File: src/models/relatedViewToggler.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+export interface RelatedViewToggler {
```
Comment (kanitw) [inline src/models/relatedViewToggler.ts]: `RelatedViews` 

Also rename the file and all other relevant things

File: src/reducers/index.test.ts [hunk 6]
```diff
@@ -76,6 +77,7 @@ describe('reducers/index', () => {
       expect(selectBookmark(state)).toEqual(DEFAULT_BOOKMARK);
       expect(selectCustomWildcardFields(state)).toEqual(DEFAULT_CUSTOM_WILDCARD_FIELDS);
       expect(selectDataset(state)).toEqual(DEFAULT_DATASET);
+      expect(selectRelatedViewToggler(state)).toEqual(DEFAULT_RELATED_VIEW_TOGGLER);
```
Comment (kanitw) [inline src/reducers/index.test.ts]: selectRelatedViews

File: src/reducers/index.ts [hunk 7]
```diff
@@ -86,21 +88,22 @@ const persistentStateToReset: ResetIndex<PersistentState> = {
...
   log: false,
+  relatedViews: true,
   shelfPreview: true
 };
 
 const persistentReducer = makeResetReducer(
   combineReducers<PersistentState>({
     bookmark: bookmarkReducer,
     config: configReducer,
+    relatedViewToggler: relatedViewReducer,
```
Comment (kanitw) [inline src/reducers/index.ts]: relatedViews

File: src/reducers/related-view.test.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        isHidden: false
+      },
+      {
+        type: RELATED_VIEWS_HIDE_TOGGLE
+      }
+    )).toEqual(expectedRelatedViews);
+  });
+});
+
+describe('reducers/related-views', () => {
```
Comment (kanitw) [inline src/reducers/related-view.test.ts]: Usually, we have just one 'describe' for multiple test cases that are under the same hierarchy

File: src/reducers/related-views.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import {Action} from '../actions/index';
+import {DEFAULT_RELATED_VIEWS, RelatedViews} from '../models/related-views';
+
+export function relatedViewsReducer(relatedViewToggler: RelatedViews = DEFAULT_RELATED_VIEWS,
+                                    action: Action): RelatedViews {
+  const {isHidden} = relatedViewToggler;
```
Comment (kanitw) [inline src/reducers/related-views.ts]: Don't you have to check for action?  Wouldn't this mean that all actions will toggle the related view?

File: src/reducers/related-views.ts [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+  return {
+    isHidden: !isHidden
```
Comment (kanitw) [inline src/reducers/related-views.ts]: 2) Why do we return `isHidden: !isHidden`, not just `isHidden`?

File: src/actions/related-views.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
+import {PlainReduxAction} from './redux-action';
+
+export type RelatedViewAction = HideRelatedViews;
```
Comment (kanitw) [inline src/actions/related-views.ts]: RelatedViewsAction

File: src/actions/index.ts [hunk 12]
```diff
@@ -77,6 +79,8 @@ const ACTION_TYPE_INDEX: {[k in ActionType]: 1} = {
   LOG_WARNINGS_ADD: 1,
   LOG_WARNINGS_CLEAR: 1,
 
+  RELATED_VIEW_HIDE_TOGGLE: 1,
```
Comment (kanitw) [inline src/actions/index.ts]: RELATED_VIEWS_HIDE_TOGGLE

File: src/reducers/related-views.ts (hunk 10 above)
Comment (kanitw) [reply src/reducers/related-views.ts]: I also wonder if just `hidden` is a better name than `isHidden`, but I don't have strong preference about this part.

Comment (felixcodes) [reply src/reducers/related-views.ts]: @kanitw Please correct me if I'm misunderstanding -- but I believe the `relatedViewToggler` parameter contains the current state before the action is pressed.  The value of `isHidden` is derived from `relatedViewToggler.isHidden` -- i.e. the current value of isHidden. `RelatedViewsAction` should flip the value of `isHidden` and returning `{isHidden}` as per your suggestion wouldn't flip the value at all.

@ssharif6 Is this what you were thinking?

Comment (ssharif6) [reply src/reducers/related-views.ts]: @felixcodes Yes. I was thinking along the lines of toggling the `isHidden` value. So a `isHidden: !isHidden` would flip from displaying hidden views in the state, and then not displaying them.

Comment (kanitw) [reply src/reducers/related-views.ts]: That make sense. Still wouldn't you need an action?

Comment (felixcodes) [reply src/reducers/related-views.ts]: Yes, @ssharif6 is missing a 
switch (action.type) case RelatedViewsAction:

Comment (felixcodes) [reply src/reducers/related-views.ts]: case RELATED_VIEWS_HIDE_TOGGLE:*** (on mobile, can't edit comments)

Comment (ssharif6) [reply src/reducers/related-views.ts]: Yes! That is definitely true and is being added for next iteration

File: src/reducers/related-views.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
+import { Action } from '../actions/index';
+import { RELATED_VIEWS_HIDE_TOGGLE } from '../actions/related-views';
+import { DEFAULT_RELATED_VIEWS, RelatedViews } from '../models/related-views';
+
+export function relatedViewsReducer(relatedViewToggler: RelatedViews = DEFAULT_RELATED_VIEWS,
+                                    action: Action): RelatedViews {
```
Comment (kanitw) [inline src/reducers/related-views.ts]: FYI, I actually don't like when a code is tabbed by half a page.  Perhaps better to tab by two or do

```
export function relatedViewsReducer(
  relatedViewToggler: RelatedViews = DEFAULT_RELATED_VIEWS,
  action: Action
): RelatedViews {
```

**Pull request merged**